-1 
Tested at: 52e8e6b27d79657152508970040fbb1dc4f0fce4
 I found a compilation error while trying to compile with clang: 
I used this command: 
 scram b vclean && scram build -k -j 32 USER_CXXFLAGS='-fsyntax-only' COMPILER='llvm compile'


 <pre>/afs/cern.ch/cms/sw/ReleaseCandidates/vol0/slc6_amd64_gcc491/cms/cmssw/CMSSW_7_5_X_2015-07-05-1100/src/DataFormats/Common/interface/OwnVector.h:417:44: warning: unused typedef 'holder_type' [-Wunused-local-typedef]
    typedef reftobase::RefHolder<ref_type> holder_type;
                                           ^
1 warning generated.
In file included from /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_5_X_2015-07-05-1100/src/HLTrigger/JetMET/src/SealModule.cc:74:
/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_5_X_2015-07-05-1100/src/HLTrigger/JetMET/src/HLTJetCollectionsForBoostedLeptonPlusJets.cc:103:25: error: unused variable 'theJetCollection' [-Werror,-Wunused-variable]
  const JetCollection & theJetCollection = *theJetCollectionHandle;
                        ^
/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_5_X_2015-07-05-1100/src/HLTrigger/JetMET/interface/HLTJetCollectionsForBoostedLeptonPlusJets.h:42:14: note: in instantiation of member function 'HLTJetCollectionsForBoostedLeptonPlusJets<reco::PFJet>::produce' requested here
    explicit HLTJetCollectionsForBoostedLeptonPlusJets(const edm::ParameterSet&);
             ^
</pre> 
 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10013/10921/summary.html

--- a/HLTrigger/JetMET/src/HLTJetCollectionsForBoostedLeptonPlusJets.cc
+++ b/HLTrigger/JetMET/src/HLTJetCollectionsForBoostedLeptonPlusJets.cc
@@ -100,7 +100,6 @@ HLTJetCollectionsForBoostedLeptonPlusJets<jetType>::produce(edm::Event& iEvent, 
   Handle<JetCollection> theJetCollectionHandle;
   iEvent.getByToken(m_theJetToken, theJetCollectionHandle);
   
-  const JetCollection & theJetCollection = *theJetCollectionHandle;
   typename JetCollection::const_iterator jet;
   
   auto_ptr<JetCollection> allSelections(new JetCollection);

